src/livekit/MediaDevicesContext.tsx

```diff
@@ -59,12 +60,26 @@ export interface MediaDevice {
   select: (deviceId: string) => void;
 }
 
-export interface MediaDevices {
+interface InputDevices {
   audioInput: MediaDevice;
-  audioOutput: MediaDevice;
   videoInput: MediaDevice;
   startUsingDeviceNames: () => void;
   stopUsingDeviceNames: () => void;
+  usingNames: boolean;
+}
+
+export interface MediaDevices extends Omit<InputDevices, "usingNames"> {
+  audioOutput: MediaDevice;
+}
+function useShowEarpiece(): boolean {
+  const [alwaysShowIphoneEarpice] = useSetting(alwaysShowIphoneEarpieceSetting);
+  const m = useMemo(
+    () =>
+      (navigator.userAgent.match("iPhone")?.length ?? 0) > 0 ||
```

Comment:
Is there a reason we can't do this here?

```suggestion
      navigator.userAgent.includes("iPhone") ||
```